inlong-sort/sort-connectors/pulsar/src/main/java/org/apache/inlong/sort/pulsar/table/DynamicPulsarDeserializationSchema.java

```diff
@@ -132,6 +140,39 @@ public void open(DeserializationSchema.InitializationContext context) throws Exc
 
     }
 
+    /**
+     * reflect get metricGroup
+     *
+     * @param context
+     * @return
+     */
+    private MetricGroup getMetricGroup(DeserializationSchema.InitializationContext context)
```

Comment:
Why not modify FlinkPulsarSource#open directly?

Reply:
@thesumery Because pulsar self metric will be affected if I modify FlinkPulsarSource#open directly. And `org.apache.flink.streaming.connectors.pulsar.FlinkPulsarSource` will need be modified if I use adminUrl pulsar.

Reply:
> Why not modify FlinkPulsarSource#open directly?

same with me, no need to reflect the metric group. It's ok if we modify the user metrics group since we build the source by ourselves